src/OXRS_LCD.cpp

```diff
@@ -178,6 +277,56 @@ void OXRS_LCD::update(void)
       _last_lcd_trigger = 0L;
     }
   }
+  
+  // turn off rx LED if timed out
+  if (_last_rx_trigger)
+  {
+    if ((millis() - _last_rx_trigger) > RX_TX_LED_ON)
+    {
+      _set_mqtt_rx_led(0);
+      _last_rx_trigger = 0L;
+    }
+  }
+ 
+  // turn off tx LED if timed out
+  if (_last_tx_trigger)
+  {
+    if ((millis() - _last_tx_trigger) > RX_TX_LED_ON)
+    {
+      _set_mqtt_tx_led(0);
+      _last_tx_trigger = 0L;
+    }
+  }
+ 
+  // check if ethernet link status has changed
+  if (_ethernet)
```

Comment:
I still think all this should live in `show_ethernet()` - and no need for `_first_call` - just have a global `_link_status` initialised to NULL so on the first loop the `tmp != _link_status` always causes an update.

Something like this;
```
void OXRS_LCD::_show_ethernet()
{
  uint8_t current_link_status = _ethernet->linkStatus();
  if (current_link_status != _link_status)
  {
    _link_status = current_link_status;

    // refresh IP on link status change
    if (_link_status == LinkON)
    {
      _show_IP(_ethernet->localIP(), 1);
    }
    else
    {
      _show_IP(IPAddress(0,0,0,0), 0);
    }

    // refresh MAC on link status changes
    byte mac[6];
    _ethernet->MACAddress(mac);
    _show_MAC(mac);
  }
}
```

Then everything is contained in this one function and we just need to call `show_ethernet()` in our `loop()`;
```
if (_ethernet) _show_ethernet();
```

Reply:
looks good. My thought processes are sometimes different from yours.

Reply:
done